web/CMakeLists.txt

```diff
@@ -1,78 +1,45 @@
-  src/utils.js
-  style/transmission-app.scss
+file(GLOB_RECURSE TRWEB_ASSETS
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/assets/img/*.png
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/assets/img/*.svg
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/assets/img/favicon.ico
 )
 
-set(TRWEB_IMGS
-  style/images/analytics.svg
-  style/images/application-x-executable.png
-  style/images/audio-x-generic.png
-  style/images/blue-turtle.png
-  style/images/checkered-flag.svg
-  style/images/chevron-down.svg
-  style/images/chevron-up.svg
-  style/images/diagram-3-fill.svg
-  style/images/files.svg
-  style/images/folder.png
-  style/images/font-x-generic.png
-  style/images/gear-fill.svg
-  style/images/horizontal-rule.svg
-  style/images/image-x-generic.png
-  style/images/lock-fill.svg
-  style/images/logo.png
-  style/images/package-x-generic.png
-  style/images/pause-circle-active.svg
-  style/images/pause-circle-idle.svg
-  style/images/play-circle-active.svg
-  style/images/play-circle-idle.svg
-  style/images/router.svg
-  style/images/search.svg
-  style/images/team.svg
-  style/images/text-x-generic.png
-  style/images/three-dots-vertical.svg
-  style/images/toolbar-close.png
-  style/images/toolbar-folder.png
-  style/images/toolbar-info.png
-  style/images/toolbar-pause.png
-  style/images/toolbar-start.png
-  style/images/turtle.png
-  style/images/up-and-down-arrows.svg
-  style/images/video-x-generic.png
+file(GLOB_RECURSE TRWEB_TMP_BUILD
+  ${CMAKE_CURRENT_SOURCE_DIR}/public_html/assets/**/*
+  ${CMAKE_CURRENT_SOURCE_DIR}/public_html/js/*
+  ${CMAKE_CURRENT_SOURCE_DIR}/public_html/*
 )
 
-add_custom_target(
-  trweb ALL
-  COMMAND ${CMAKE_COMMAND} -E copy "${CMAKE_CURRENT_SOURCE_DIR}/package.json" "${CMAKE_CURRENT_BINARY_DIR}/package.json"
-  COMMAND ${CMAKE_COMMAND} -E copy "${CMAKE_CURRENT_SOURCE_DIR}/yarn.lock" "${CMAKE_CURRENT_BINARY_DIR}/yarn.lock"
-  COMMAND yarn install
-  COMMAND yarn webpack --config "${CMAKE_CURRENT_SOURCE_DIR}/webpack.config.js" --context "${CMAKE_CURRENT_SOURCE_DIR}"
-  BYPRODUCTS
-    public_html/transmission-app.js
-    public_html/transmission-app.js.LICENSE.txt
-  DEPENDS ${TRWEB_IMGS}
-  SOURCES ${TRWEB_SRCS}
+file(GLOB_RECURSE TRWEB_SRCS
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/*.js
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/index.html
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/assets/css/*.scss
 )
+
+find_program(NODE node)
+
+if (NODE)
+    add_custom_target(
+      trweb ALL
+      COMMENT "Building Web"
+      COMMAND node --version
+      COMMAND npm config set package-lock false
+      COMMAND npm install --no-save corepack
+      COMMAND npx corepack enable
+      COMMAND npx yarn set version stable
+      COMMAND npx yarn install WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+      COMMAND npx yarn webpack --config "${CMAKE_CURRENT_SOURCE_DIR}/webpack.config.js" --context "${CMAKE_CURRENT_SOURCE_DIR}"
+      COMMAND ${CMAKE_COMMAND} -E copy_directory "${CMAKE_CURRENT_SOURCE_DIR}/public_html" "${CMAKE_CURRENT_BINARY_DIR}/public_html"
```

Comment:
This looks like there might be cumulative effect when subsequent builds copy files over and over again. Consider removing `${CMAKE_CURRENT_BINARY_DIR}/public_html` before the copy so that it only contains files from the last build in the end, if you even need this copy operation in the first place (I'm not convinced yet).

Reply:
Hi, @mikedld. We do not need to copy the output to the global build dir. I have added this to consistently keep all build output under one location but since we actually have to compile the web in the src dir (because of yarn/node limitations) I guess we can also keep it's output there. 
Btw. public_html is cleaned before every build by a WebPack plugin.